### PR TITLE
Quick preview of Job error messages

### PR DESCRIPTION
**Pull request by chrisvanrun**
Closes: #3405

https://github.com/comic/grand-challenge.org/assets/7871310/ca188345-4fe4-4290-9ea7-d6f49f7cf14e

### CODE REVIEW

**Comment by jmsmkn** [comment]
What does this look like for algorithms that succeed but have warnings (e.g. output on stderr)?

**Comment by chrisvanrun** [comment]
image

Not sure what the interaction is between stderr and error_message. Sometimes one is deduced from the other I think? If that is the case it will show the 'warning' type but include a modal option.

**Comment by jmsmkn** [comment]
`error_message` has a bunch of sources but it will not be set if the job is successful. If a job really failed the Results column would be empty. Would it be a good idea to put the error message there rather than "Empty"+modal?

Screenshot 2024-07-04 at 13 47 55

**Comment by chrisvanrun** [comment]
With the latest commit, show casing a maximum sized `error_message` (1024 char):

https://github.com/comic/grand-challenge.org/assets/7871310/38e3bfda-544c-4a30-bf4d-c1d5f6d6a796

File: app/grandchallenge/algorithms/templates/algorithms/job_list_row.html [hunk 1]
```diff
@@ -27,13 +27,14 @@
...
-    {{ object.get_status_display }}{% if object.status == object.SUCCESS and object.stderr %},
-        with warnings{% endif %}
-</span>
+    <span class="badge badge-{{ object.status_context }}">
+        {% if object.animate %}
+            <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
+        {% endif %}
+        {{ object.get_status_display }}{% if object.status == object.SUCCESS and object.stderr %},
+            with warnings{% endif %}
+    </span>
```
Comment (jmsmkn) [inline app/grandchallenge/algorithms/templates/algorithms/job_list_row.html:36]:
```suggestion
<span class="badge badge-{{ object.status_context }}">
    {% if object.animate %}
        <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
    {% endif %}
    {{ object.get_status_display }}{% if object.status == object.SUCCESS and object.stderr %},
        with warnings{% endif %}
</span>
```

File: app/grandchallenge/algorithms/templates/algorithms/job_list_row.html [hunk 2]
```diff
@@ -54,13 +55,66 @@
 {{ object.comment }}
 <split></split>
 
+
+
```
Comment (jmsmkn) [inline app/grandchallenge/algorithms/templates/algorithms/job_list_row.html:59]:
```suggestion
```

File: app/grandchallenge/algorithms/templates/algorithms/job_list_row.html [hunk 3]
```diff
@@ -34,6 +34,7 @@
     {{ object.get_status_display }}{% if object.status == object.SUCCESS and object.stderr %},
         with warnings{% endif %}
 </span>
+
```
Comment (jmsmkn) [inline app/grandchallenge/algorithms/templates/algorithms/job_list_row.html:37]:
```suggestion
```

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**